### ISSUE TITLE
Decouple `TokenTransfer{Validation,Execution}` from `SendPacket{Validation,Execution}`

### ISSUE THREAD

**Issue by plafer** [cosmos/ibc-rs#786]
They are currently [coupled](https://github.com/cosmos/ibc-rs/blob/d6da3ed917a41cea1711bc2905dba8b4bcf6696c/crates/ibc/src/applications/transfer/context.rs#L29). However, basecoin shows us that this no longer works, since the `TransferModule` must know how to send packets, which now requires [access to the client validation context](https://github.com/cosmos/ibc-rs/blob/d6da3ed917a41cea1711bc2905dba8b4bcf6696c/crates/ibc/src/core/ics04_channel/context.rs#L30) (implemented by the core `Ibc` struct).

Basically, decoupling them just gives us more flexibility; they never *needed* to be coupled, it was just convenient.

**Comment by Farhad-Shabani** [comment]
I just noticed this issue in essence is the same matter surfaced in #520. That can be closed as well.
